tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCoverageLevel.java

```diff
@@ -1391,4 +1395,122 @@ public NumPathCoverageItem(
+        // are ordered we can test
+        // pairs
+
+        for (String locale : List.of("en" /*, "de"*/)) {
+            CLDRFile cldrFile = CLDRConfig.getInstance().getCldrFactory().make(locale, true);
+            CoverageLevel2 coverageLevel = CoverageLevel2.getInstance(locale);
+
+            Multimap<Level, PathHeader> sorted = TreeMultimap.create();
+            for (String path : cldrFile) {
+                if (!path.startsWith("//ldml/dates/calendars/calendar[@type=\"iso8601\"]")
+                        || path.endsWith("/alias")) {
+                    continue;
+                }
+                Level actual = coverageLevel.getLevel(path);
+                PathHeader ph = PathHeader.getFactory().fromPath(path);
+                ph.getPageId();
+                if (ph.getPageId() != PageId.Suppress) {
+                    sorted.put(actual, ph);
+                }
+            }
+            Set<String> ids = new TreeSet<>();
+            Set<String> keepIdSet = new TreeSet<>();
+            for (Entry<Level, Collection<PathHeader>> entry : sorted.asMap().entrySet()) {
+                Level level = entry.getKey();
+
+                for (PathHeader ph : entry.getValue()) {
+
+                    String path = ph.getOriginalPath();
+                    XPathParts parts = XPathParts.getFrozenInstance(path);
+
+                    // account for .../intervalFormatItem[@id="Gy"]/greatestDifference[@id="G"]
+                    // and for .../dateFormatItem[@id="Gy"]
+                    String id = parts.getAttributeValue(-2, "id");
+                    if (id == null) {
+                        id = parts.getAttributeValue(-1, "id");
+                    }
+                    if (id != null) {
+                        ids.add(id);
+                    }
+
+                    boolean expectedComprehensive =
+                            comprehesiveElements.stream().anyMatch(x -> containing(parts, x));
+                    if (!expectedComprehensive) {
+                        if (path.endsWith("/pattern[@type=\"standard\"]")
+                                && !path.contains("/dateFormat[@type=\"standard\"]")) {
+                            expectedComprehensive = true;
+                        } else if (id != null) {
+                            boolean keepIdS = keepIds.contains(id);
+                            if (keepIdS) {
+                                keepIdSet.add(id);
+                            } else {
+                                expectedComprehensive = !keepIdS;
+                                boolean keepIdR = okIdPattern.matcher(id).matches();
+                                if (keepIdS != keepIdR) {
+                                    throw new IllegalArgumentException("ID mismatch: " + id);
+                                }
+                            }
+                        }
+                    }
+                    if (Level.CORE_TO_MODERN.contains(level) == expectedComprehensive) {
```

Comment:
```suggestion
                    if (!level.isAbove(MODERN) == expectedComprehensive) {
```

Reply:
Good suggestion but I didn't want to wait on getting PR in.